azurerm/internal/services/loganalytics/log_analytics_linked_service_resource.go

```diff
@@ -167,6 +167,31 @@ func resourceArmLogAnalyticsLinkedService() *schema.Resource {
 				}
 			}
 
+			// TODO: Remove in 3.0
+			if d.HasChange("resource_id") {
+				if resourceID := d.Get("resource_id").(string); resourceID != "" {
+					if !validate.IsAutomationAccountID(resourceID) {
+						return fmt.Errorf("'resource_id' must be an Automation Account resource ID, got %q", resourceID)
+					}
+				}
+			}
+
+			if d.HasChange("read_access_id") {
+				if readAccessID := d.Get("read_access_id").(string); readAccessID != "" {
+					if !validate.IsAutomationAccountID(readAccessID) {
+						return fmt.Errorf("'read_access_id' must be an Automation Account resource ID, got %q", readAccessID)
+					}
+				}
+			}
+
+			if d.HasChange("write_access_id") {
+				if writeAccessID := d.Get("write_access_id").(string); writeAccessID != "" {
+					if !validate.IsLogAnalyticsClusterID(writeAccessID) {
```

Comment:
There's a generated validator for this one in loganalytics validate, maybe use that one instead?

Reply:
Fair point... I'll update my logic to use the generated validator rather than my own.